src/ui/ServerAdministration/UsersTable/UserSidebar/UserStatusRoleTable.tsx

```diff
@@ -0,0 +1,115 @@
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import * as React from 'react';
+
+import {
+  ButtonDropdown,
+  DropdownItem,
+  DropdownMenu,
+  DropdownToggle,
+} from 'reactstrap';
+import { ISelection } from '..';
+import * as ros from '../../../../services/ros';
+
+export interface IUserStatusRoleTableProps {
+  selection: ISelection;
+  roleDropdownOpen: boolean;
+  toggleRoleDropdown: () => void;
+  onRoleChanged: (role: ros.UserRole) => void;
+  onStatusChanged: (status: ros.UserStatus) => void;
+  statusDropdownOpen: boolean;
+  toggleStatusDropdown: () => void;
+}
+
+export const UserStatusRoleTable = ({
+  selection,
+  roleDropdownOpen,
+  toggleRoleDropdown,
+  onRoleChanged,
+  onStatusChanged,
+  statusDropdownOpen,
+  toggleStatusDropdown,
+}: IUserStatusRoleTableProps) => (
+  <table>
+    <tbody>
+      <tr>
+        <td>Role</td>
+        <td>
+          <ButtonDropdown isOpen={roleDropdownOpen} toggle={toggleRoleDropdown}>
+            <DropdownToggle caret={true}>
+              {selection.user.isAdmin ? 'Administrator' : 'Regular user'}
+            </DropdownToggle>
+            <DropdownMenu>
+              <DropdownItem
+                onClick={() => onRoleChanged(ros.UserRole.Administrator)}
+              >
+                Administrator
+              </DropdownItem>
+              <DropdownItem onClick={() => onRoleChanged(ros.UserRole.Regular)}>
+                Regular user
+              </DropdownItem>
+            </DropdownMenu>
+          </ButtonDropdown>
+        </td>
+      </tr>
+      <tr>
+        <td>Status</td>
+        <td>
+          <ButtonDropdown
+            isOpen={statusDropdownOpen}
+            toggle={toggleStatusDropdown}
+          >
+            <DropdownToggle caret={true}>
+              {humanizeUserStatus(selection.user.status)}
+            </DropdownToggle>
+            <DropdownMenu>
+              {[ros.UserStatus.active, ros.UserStatus.blacklisted].map(
+                status => {
+                  return (
+                    <DropdownItem
+                      key={status}
+                      onClick={() => onStatusChanged(status)}
+                    >
+                      {humanizeUserStatus(status)}
+                    </DropdownItem>
+                  );
+                },
+              )}
+            </DropdownMenu>
+          </ButtonDropdown>
+        </td>
+      </tr>
+    </tbody>
+  </table>
+);
+
+function humanizeUserStatus(userStatus: ros.UserStatus | string) {
```

Comment:
```suggestion
function humanizeUserStatus(userStatus: ros.UserStatus | string = ros.UserStatus.active) {
```

How about making active the default value of the argument instead of listing `'', null and undefined`?

Reply:
I need to check how defaults behave when you pass in `null`. E.g. would `humanizeUserStatus(null)` set `userStatus` to `active` or will it pass `null`?

Reply:
It appears that only `undefined` will get the default value, but `humanizeUserStatus("")` will set `userStatus` to the empty string and `humanizeUserStatus(null)` will set it to `null`.